spec/models/restaurant_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'rails_helper'
+
+RSpec.describe Restaurant, type: :model do
+  let(:rest) do
+    {
+      name: 'Name',
+      genre: :sushi,
+      rating: 1,
+      accepts_10bis: true,
+      delivery_time: :slow
+    }
+  end
+
+  describe 'validations' do
+    it 'should require all fields except address' do
+      rest.keys.each do |field|
+        bad_rest = rest.select { |k, _v| k != field }
+        expect(Restaurant.new(bad_rest).valid?).to be false
+      end
+    end
+
+    it 'should validate name length' do
```

Comment:
Very useful for testing validations: http://www.rubydoc.info/github/thoughtbot/shoulda-matchers/Shoulda%2FMatchers%2FActiveModel%3Avalidate_presence_of